controllers/history.controller.js

```diff
@@ -0,0 +1,81 @@
+const userDb = require("../models/user.model");
+
+module.exports.getHistory = async (req, res) => {
+  const { userId } = req.params;
+  try {
+    const user = await userDb.findById(userId);
+    const data = await user.execPopulate("history");
+    return res.status(200).json({ success: true, data: [...data.history] });
+  } catch (error) {
+    return res
+      .status(500)
+      .json({ success: false, message: "internal server error" });
+  }
+};
+
+module.exports.addToHistory = async (req, res) => {
+  const { userId, videoId } = req.params;
+  try {
+    const user = await userDb.findById(userId);
+    const history = user.history;
+    if (!history.some((video) => video == videoId)) {
+      await user.history.push(videoId);
```

Comment:
`await` for push can be removed.  Also, you have `history` available above.  So, `history.push(videoId)` can be written